src/index/TextIndexReadWrite.cpp

```diff
@@ -4,6 +4,67 @@
 
 #include "index/TextIndexReadWrite.h"
 
+#include "TextScoringEnum.h"
+
+namespace textIndexReadWrite::detail {
+
+// _____________________________________________________________________________
+IdTable readContextListHelper(
+    const ad_utility::AllocatorWithLimit<Id>& allocator,
+    const ContextListMetaData& contextList, bool isWordCl,
+    const ad_utility::File& textIndexFile,
+    TextScoringMetric textScoringMetric) {
+  IdTable idTable{3, allocator};
+  idTable.resize(contextList._nofElements);
+  // Read ContextList
+  readGapComprList<Id, uint64_t>(
+      idTable.getColumn(0).begin(), contextList._nofElements,
+      contextList._startContextlist, contextList.getByteLengthContextList(),
+      textIndexFile, [](uint64_t id) {
+        return Id::makeFromTextRecordIndex(TextRecordIndex::make(id));
+      });
+
+  // Helper lambda to read wordIndexList
+  auto wordIndexToId = [isWordCl](auto wordIndex) {
+    if (isWordCl) {
+      return Id::makeFromWordVocabIndex(WordVocabIndex::make(wordIndex));
+    }
+    return Id::makeFromVocabIndex(VocabIndex::make(wordIndex));
+  };
+
+  // Read wordIndexList
+  readFreqComprList<Id, WordIndex>(
+      idTable.getColumn(1).begin(), contextList._nofElements,
+      contextList._startWordlist, contextList.getByteLengthWordlist(),
+      textIndexFile, wordIndexToId);
+
+  // Helper lambdas to read scoreList
+  auto scoreToId = []<typename T>(T score) {
```

Comment:
We are refactoring to C++ 17, so no templated lambdas:(
use `auto score` and then `using T = decltype(score)` in the lambda body.